File: json.scala
import scala.util.Random

object RanCode {
  val chars = asciiLetters + digits

  def ranCode(length: Int): String = {
  (0 until length).map(_ => Random.choice(chars)).mkString
  }

  def fileValue(name: String, exp: String, data: Map[String, Any]): Unit = {
  val file = new java.io.FileWriter(name + "." + exp)
  java.json.JSONObject.writeTo(file, data)
  }

  def main(args: Array[String]) {
  println("Введите имя - ")
  val username = scala.io.StdIn.readLine()
  val password = ranCode(Random.nextInt(10) + 10)
  val data = Map(
  "username" -> username,
  "password" -> password,
  "iflogin?" -> true
  )
  fileValue("File", "json", data)
  }
}

File: json.kt
import java.util.Random

fun generatePassword(length: Int): String {
  val chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
  return StringBuilder(length).appendRandom(chars).toString()
}

fun saveToFile(name: String, extension: String, data: Map<String, Any>) {
  val file = File("$name.$extension")
  file.writer().use { writer ->
  writer.write(JSON.stringify(data))
  }
}

fun main() {
  val username = readLine("Введите имя - ")
  val password = generatePassword(random.nextInt(10) + 10)
  val data = mapOf(
  "username" to username,
  "password" to password,
  "iflogin?" to true
  )
  saveToFile("File", "json", data)
}